src/main/java/nl/knaw/dans/layerstore/LayerManagerImpl.java

```diff
@@ -40,37 +38,42 @@ public class LayerManagerImpl implements LayerManager {
     @Getter
     private Layer topLayer;
 
-    public LayerManagerImpl(@NonNull Path stagingRoot, @NonNull ArchiveProvider archiveProvider, Executor archivingExecutor) {
+    public LayerManagerImpl(@NonNull Path stagingRoot, @NonNull ArchiveProvider archiveProvider, Executor archivingExecutor) throws IOException {
         this.stagingRoot = stagingRoot;
         this.archivingExecutor = Objects.requireNonNullElseGet(archivingExecutor, Executors::newSingleThreadExecutor);
         this.archiveProvider = archiveProvider;
         initTopLayer();
     }
 
-    public LayerManagerImpl(@NonNull Path stagingRoot, @NonNull ArchiveProvider archiveProvider) {
+    public LayerManagerImpl(@NonNull Path stagingRoot, @NonNull ArchiveProvider archiveProvider) throws IOException {
         this(stagingRoot, archiveProvider, null);
     }
 
-    @SneakyThrows
-    private void initTopLayer() {
-        List<Path> paths;
-        try (Stream<Path> pathStream = Files.list(stagingRoot)) {
-            paths = pathStream.toList();
+    private void initTopLayer() throws IOException {
+        if (Files.notExists(stagingRoot)) {
```

Comment:
Not sure about whether auto-creating the staging root can cause surprises, but I can't think of a concrete scenario, so I'll go along with your change.